### DISCUSSION TITLE
Oauth/OpenID authentication

### DISCUSSION THREAD

**Discussion by simse** [Budibase/budibase#1630] [Ideas]
**Reason**
For enterprise users, the lack of any general authentication (SAML, Oauth or OpenID) method is a deal-breaker. I cannot even begin to explore the project, because it is undeployable. :( 

**Problem**
It locks out a large market segment from using this project.

**Fix**
Add general Oauth support. Oauth is also useful for non-enterprise users seeking to integrate with unsupported Oauth providers or internal auth set ups.

IF I can help, I will be happy to. I am proficient in Javascript.

**Comment by Hokwang** [comment]
yes, this is must feature.
I prefer OpenID and now use keycloak for that.

**Reply by burnoutberni** [reply]
Same, I think OIDC support would be super helpful, maybe we can get this working #1806

**Comment by mjashanks** [comment]
Agreed, this is very important.

@simse and @Hokwang I'd love to have a call with you sometime if you are available, to gather more about what you need around these features. My email is in my profile - would really appreciate it :)

**Comment by timchkor** [comment]
Yeah... I Agree! Those SSO is must have feature..

**Reply by mjashanks** [reply]
@timchkor do you have a preference for an SSO implementation?

**Reply by timchkor** [reply]
Yeah, integration with MS AAD - SSO would be the major considerations for internal users

**Comment by Manishkparmar80** [comment]
Also consider LDAP integration with microsoft AD for SSO functionality

**Reply by joebudi** [reply]
We believe this will be a common requirement. We will bring it to our roadmap meeting.

**Reply by Manishkparmar80** [reply]
@joebudi Thanks for understanding. This will suerly do value addition to Budibase application and would be useful for most of enterprises. Looking forward for positive outcome of your roadmap meeting.

**Reply by masykur** [reply]
@joebudi I prefer to support OpenID so we are able to integrate to any OpenID supported provider.

**Comment by titoanugerah** [comment]
On Budibase version 0.9.83, its look like budibase support for OIDC, but after I try to connect using keycloak appear message "Error constructing OIDC authentication strategy". 
![image](https://user-images.githubusercontent.com/35187235/127269085-a4bf7a45-c636-4ea2-98e7-9c9b244efdbe.png)
![image](https://user-images.githubusercontent.com/35187235/127269226-ff057294-3564-446f-abbd-0a0e7d738593.png)

**Reply by Rory-Powell** [reply]
Hey @titoanugerah, sorry you're having issues. The error message isn't very helpful here. 
Before continuing can you have a look through these docs https://docs.budibase.com/admin/authentication 
If you're still having issues we can dig a little further

**Reply by masykur** [reply]
Hi @titoanugerah,
Budibase need full config url instead of just auth server url or auth server url + realm.
Pease set config url as below
`http[s]://keycloak-domain.com/auth/realms/{change-to-your-realm}/.well-known/openid-configuration`
**Note**: Client ID is case sensitive, please set to correct case

**Reply by PClmnt** [reply]
Some Keycloak docs have been added: https://docs.budibase.com/admin/authentication/sso-with-keycloak

**Reply by artemevskikh** [reply]
Hi, my budibase version 1.0.90. But I'm getting a similar error. I want to login with Keycloak. Tell me, what's the problem?
![image](https://user-images.githubusercontent.com/49155124/160136749-67c86b48-f400-4496-b7b5-8e80ecf470c6.png)

**Reply by MartinPicc** [reply]
Any update on this issue ? 
I'm still getting this error with Budibase 1.2.36 running locally on docker and Keycloak 19.0.2 (also tried 18.0.2)

**Comment by Rory-Powell** [comment]
OIDC support has been shipped in version 0.9.83 🎉 
Check out the documentation for more details: https://docs.budibase.com/admin/authentication

**Reply by itsthisjustin** [reply]
FYI this is now a 404

**Comment by marco-santulli** [comment]
do we still have plan for SAML / AD integrations for SSO?
